device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/qos.json.j2

```diff
@@ -100,109 +103,207 @@
-        "scheduler.8": {
-            "type"  : "DWRR",
+        "scheduler_q1_downlink": {
+            "type": "DWRR",
             "weight": "8"
         },
-        "scheduler.18": {
-            "type"  : "DWRR",
+        "scheduler_q2_downlink": {
+            "type": "DWRR",
             "weight": "18"
         },
-        "scheduler.22": {
-            "type"  : "DWRR",
+        "scheduler_q3_downlink": {
+            "type": "DWRR",
             "weight": "22"
-        }
-    },
-{%- endmacro -%}
-
-{%- macro generate_single_queue_per_sku(port) -%}
-        "{{ port }}|0": {
-            "scheduler": "scheduler.4"
         },
-        "{{ port }}|1": {
-            "scheduler": "scheduler.8",
-            "wred_profile": "AZURE_LOSSY_Q1"
+        "scheduler_q4_downlink": {
+            "type": "DWRR",
+            "weight": "22"
         },
-        "{{ port }}|2": {
-            "scheduler": "scheduler.18",
-            "wred_profile": "AZURE_LOSSY_Q2"
+        "scheduler_q5_downlink": {
+            "type": "DWRR",
+            "weight": "22"
         },
-        "{{ port }}|3": {
-            "scheduler": "scheduler.22",
-            "wred_profile": "AZURE_LOSSY_Q3"
+        "scheduler_q6_downlink": {
+            "type": "DWRR",
+            "weight": "4"
         },
-        "{{ port }}|4": {
-            "scheduler": "scheduler.22"
+        "scheduler_q0_uplink": {
+            "type": "DWRR",
+            "weight": "4"
         },
-        "{{ port }}|5": {
-            "scheduler": "scheduler.22"
+        "scheduler_q1_uplink": {
+            "type": "DWRR",
+            "weight": "8"
         },
-        "{{ port }}|6": {
-            "scheduler": "scheduler.4"
+        "scheduler_q2_uplink": {
+            "type": "DWRR",
+            "weight": "18"
+        },
+        "scheduler_q3_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q4_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q5_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q6_uplink": {
+            "type": "DWRR",
+            "weight": "4"
         }
+    },
+{%- endmacro -%}
+
+{%- macro generate_direction_based_queue_per_sku(port, direction) -%}
+    "{{ port }}|0": {
+        "scheduler": "scheduler_q0_{{ direction }}"
+    },
+    "{{ port }}|1": {
+        "scheduler": "scheduler_q1_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q1"
+    },
+    "{{ port }}|2": {
+        "scheduler": "scheduler_q2_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q2"
+    },
+    "{{ port }}|3": {
+        "scheduler": "scheduler_q3_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q3"
+    },
+    "{{ port }}|4": {
+        "scheduler": "scheduler_q4_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q4"
```

Comment:
WRED profile isn't defined for q4 and q5.
Could you please define them?
According to the document, they are not enabled for now, and we just need a placeholder.
Please come up with a way to define it in SONiC.

Reply:
Closed